desc/coils.py

```diff
@@ -1328,6 +1481,756 @@ def _check_type(coil0, coil):
+
+        Returns
+        -------
+        vec_proj : ndarray, shape(n,3)
+            Vector reprojected into local coil xsection frame at each point of the
+            centerline grid.
+
+        """
+        t_frame = self.compute(
+            "centroid_tangent",
+            grid=centerline_grid,
+            params=params,
+            basis="xyz",
+        )["centroid_tangent"]
+        p_frame = self.compute(
+            "p_frame",
+            grid=centerline_grid,
+            params=params,
+            basis="xyz",
+        )["p_frame"]
+        q_frame = self.compute(
+            "q_frame",
+            grid=centerline_grid,
+            params=params,
+            basis="xyz",
+        )["q_frame"]
+
+        t_frame = finite_build_grid.expand(t_frame, "zeta")
+        p_frame = finite_build_grid.expand(p_frame, "zeta")
+        q_frame = finite_build_grid.expand(q_frame, "zeta")
+
+        vec_t = dot(vec, t_frame)
+        vec_p = dot(vec, p_frame)
+        vec_q = dot(vec, q_frame)
+        vec_proj = jnp.stack((vec_t, vec_p, vec_q), axis=-1)
+
+        return vec_proj
+
+
+class FourierPlanarFiniteBuildCoil(_FiniteBuildCoil, FourierPlanarCoil):
+    """Coil that lies in a plane, with a finite cross section.
+
+    Refer to FourierPlanarCoil for a description of the parameterization for the
+    planar coil centerline. In the case of rectangular cross sections, the coil
+    cross section is assumed to remain aligned with respect to the coil plane.
+    The first dimension of the rectangular cross section is within the coil plane,
+    and the second dimension is in the coil plane normal direction.
+    No twist angle is assumed for this coil.
+
+    Parameters
+    ----------
+    cross_section_dims : array-like
+        Dimensions of the coil cross section, with 1 or 2 dimensions depending on
+        the cross section shape (circular or rectangular).
+    cross_section_shape : str
+        Shape of the coil cross section, either 'circular' or 'rectangular'.
+    current : float
+        Current through the coil, in Amperes.
+    center : array-like, shape(3,)
+        Coordinates of center of curve, in system determined by basis.
+    normal : array-like, shape(3,)
+        Components of normal vector to planar surface, in system determined by basis.
+    r_n : array-like
+        Fourier coefficients for radius from center as function of polar angle
+    modes : array-like
+        mode numbers associated with r_n
+    basis : {'xyz', 'rpz'}
+        Coordinate system for center and normal vectors. Default = 'xyz'.
+    name : str
+        Name for this coil.
+    """
+
+    _io_attrs_ = _FiniteBuildCoil._io_attrs_ + FourierPlanarCurve._io_attrs_
+
+    def __init__(
+        self,
+        cross_section_dims=[0.1, 0.1],
+        cross_section_shape="rectangular",
+        current=1,
+        center=[10, 0, 0],
+        normal=[0, 1, 0],
+        r_n=2,
+        modes=None,
+        basis="xyz",
+        name="",
```

Comment:
minor point but i'd prefer if the ordering kept the regular FourierPlanarCoil params first and added the finite build params on at the end

Reply:
Didn't we decide that all new classes should require keyword arguments going forward? So let's enforce that by adding `*` after `self`.  (And also make this change for all the other new classes added in this PR.)

Reply:
that was more for objectives, I don't think we decided for that for all classes